### PR TITLE
CIFAR10 sample

### PR DESCRIPTION
**Pull request by RodionovF**

### CODE REVIEW

File: samples/cifar10/main.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            model.forward(X, predict);
+            acc += dw::metric::accuracy(predict, y);
+            ++val_iter;
+        }
+
+        acc /= val_iter;
+        std::cout << "Accuracy: " << acc << std::endl;
+    }
+
+    std::cout << "Model saved: mnist_model.bin" << std::endl;
```
Comment (l-bat) [inline samples/cifar10/main.cpp:126]:
```suggestion
    std::cout << "Model saved: cifar10_model.bin" << std::endl;
```

File: src/runtime/tensor.cpp [hunk 2]
```diff
@@ -94,6 +94,11 @@ void dw::Tensor::allocate(const dw::Shape& shape) {
     m_descriptor->data = m_descriptor->memory.get();
 }
 
+void dw::Tensor::reshape(const dw::Shape& shape) {
+    m_descriptor->shape = shape;
```
Comment (l-bat) [inline src/runtime/tensor.cpp:98]: Please add check` tensor.total() == shape.total()`

File: samples/cifar10/main.cpp (hunk 1 above)
Comment (RodionovF) [reply samples/cifar10/main.cpp]: Thanks!
done

File: src/runtime/tensor.cpp (hunk 2 above)
Comment (RodionovF) [reply src/runtime/tensor.cpp:101]: added

**Comment by RodionovF** [comment]
architecture reached 0.58 accuracy in 20 epochs with a batch size == 64

in the future, we can add batchnorm2d to this architecture, etc

the test will be in another pull request

**Review by KSTconsult — APPROVED** [review]

**Review by TolyaTalamanov — APPROVED** [review]
@l-bat Does it make sense to add more convolution/pooling layers to this topology ?

File: samples/mnist/main.cpp [hunk 3]
```diff
@@ -23,8 +23,8 @@ static dw::Model buildMNISTModel(int batch_size) {
 int main(int argc, char *argv[]) {
     // Configuration
     std::string root = argv[1];
-    auto train_dir   = root + "train";
-    auto test_dir    = root + "test";
+    auto train_dir   = root + "/train";
```
[the remark this answers is not in the document]
Comment (KSTconsult) [reply samples/mnist/main.cpp:26]: I suppose it`s my fault. Need to replace run instruction in readme from ./bin/sample_cifar10_train /tmp/CIFAR10 to ./bin/sample_cifar10_train /tmp/CIFAR10/

[the remark this answers is not in the document]
Comment (KSTconsult) [reply samples/mnist/main.cpp:26]: Or here like  Fedor suggests

**Comment by l-bat** [comment]
> architecture reached 0.58 accuracy in 20 epochs with a batch size == 64
 Maybe add these results to README?

**Comment by l-bat** [comment]
> @l-bat Does it make sense to add more convolution/pooling layers to this topology ?

I think we can add more convolution/pooling layers to CIFAR100 sample

**Comment by TolyaTalamanov** [comment]
Let me know when it's ready to be merged

**Comment by TolyaTalamanov** [comment]
@RodionovF Is there any test for this network in `test_networks.cpp` ?

**Review by l-bat — APPROVED** [review]

File: tests/unit/test_networks.cpp [hunk 4]
```diff
@@ -360,3 +360,303 @@ TEST_F(MNISTModel, SaveLoadParams) {
...
+    int image_size   = 32;
+    int out_features = 10;
+
+    std::array<int, 2> kernel_conv{5, 5};
+    std::array<int, 2> padding_conv{2, 2};
+    std::array<int, 2> stride_conv{1, 1};
+
+    std::array<int, 2> kernel_pool{1, 1};
+    std::array<int, 2> padding_pool{0, 0};
+    std::array<int, 2> stride_pool{1, 1};
```
Comment (l-bat) [inline tests/unit/test_networks.cpp:513]:
```suggestion
    std::array<int, 2> kernel_pool{2, 2};
    std::array<int, 2> padding_pool{0, 0};
    std::array<int, 2> stride_pool{2, 2};
```

**Review by l-bat — APPROVED** [review]

**Comment by TolyaTalamanov** [comment]
Great! Thanks :+1:
